Extensions/Firebase/firebasetools.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Firebase Tools
+ * @author arthuro555
+ */
+
+/*    Initialization of the firebase SDK    */
+
+/** A Timeout function is used to initialize firebase as soon as the project data has been loaded */
+let gdjsWatcher = function() {
+    if (gdjs.projectData === undefined) {
+        setTimeout(gdjsWatcher, 1);
+        return;
+    }
+    let firebaseConfig;
+    try { 
+        firebaseConfig = JSON.parse(gdjs.projectData.properties.firebaseConfig);
+    } catch {
+        console.error("The Firebase configuration is invalid!");
+    }
+    firebase.initializeApp(firebaseConfig);
+};
+gdjsWatcher();
+
+/*    Creation of all Namespaces    */
+
+/**
+ * Firebase Event Tools
+ * @namespace
+ */
+gdjs.evtTools.firebase = {};
+
+/**
+ * Firebase Analytics Tools
+ * @namespace
+ */
+gdjs.evtTools.firebase.analytics = {};
```

Comment:
I will probably move every separate module of the firebase SDK to their own "tools.js" to prevent exporting useless code

Reply:
That sounds fine 👍